### PR TITLE
fix: include commas in the profile size, handle failure case of profile more than 500kB in size

### PR DESCRIPTION
**Pull request by yashasvibajpai**
# Description

On debugging for the issue, we came across the two pointers addressed here.
1. commas in JSON were unaccounted for in the final size of payload.
2. we will throw an error in case a single profile's size itself crosses the 500 kb threshold set by Klaviyo themselves. [Reference](https://developers.klaviyo.com/en/docs/use_klaviyos_bulk_profile_import_api#limitations)

## Linear Ticket

Related to INT-2820, INT-2840

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: router/batchrouter/asyncdestinationmanager/klaviyobulkupload/klaviyobulkupload.go
[the remark this answers is not in the document]
Comment (yashasvibajpai) [reply router/batchrouter/asyncdestinationmanager/klaviyobulkupload/klaviyobulkupload.go:1]: Handling them now in separate PRs, this PR is mainly a followup to a reported issue

**Review by koladilip — APPROVED** [review]

File: router/batchrouter/asyncdestinationmanager/klaviyobulkupload/klaviyobulkupload.go [hunk 1]
```diff
@@ -333,6 +338,15 @@ func (kbu *KlaviyoBulkUploader) Upload(asyncDestStruct *common.AsyncDestinationS
 			return kbu.generateKlaviyoErrorOutput("Error while parsing JSON.", err, importingJobIDs, destinationID)
 		}
 		profileStructure := kbu.ExtractProfile(input)
+		// if profileStructure length is more than 5 mb, throw an error
```
Comment (krishna2020) [inline router/batchrouter/asyncdestinationmanager/klaviyobulkupload/klaviyobulkupload.go:341]:
```suggestion
		// if profileStructure length is more than 500 kB, throw an error
```

File: router/batchrouter/asyncdestinationmanager/klaviyobulkupload/klaviyobulkupload.go
Comment (krishna2020) [inline router/batchrouter/asyncdestinationmanager/klaviyobulkupload/klaviyobulkupload.go:1]: Add test cases

File: router/batchrouter/asyncdestinationmanager/klaviyobulkupload/klaviyobulkupload.go (hunk 1 above)
Comment (yashasvibajpai) [reply router/batchrouter/asyncdestinationmanager/klaviyobulkupload/klaviyobulkupload.go]: updated

File: router/batchrouter/asyncdestinationmanager/klaviyobulkupload/klaviyobulkupload.go
Comment (yashasvibajpai) [reply router/batchrouter/asyncdestinationmanager/klaviyobulkupload/klaviyobulkupload.go:1]: added

**Pull request closed without merging**